### PR TITLE
Validate tail_acl and segments_vlans_file during Forch initialization

### PR DESCRIPTION
**Pull request by didovesei**
The main idea of this change is that if either of these two configs are invalid (specifed acl/file does not existing, file reading error, etc.), Forch will continue to run with static and dynamic auth disabled. Forch will also display the detailed error on NOAH.

### CODE REVIEW

File: forch/forchestrator.py [hunk 1]
```diff
@@ -220,10 +221,27 @@ def _initialize_orchestration(self):
             self._initialize_faucetizer(sequester_segment)
             self._faucetizer.reload_structural_config()
 
+            tail_acl = self._config.orchestration.tail_acl
+            if tail_acl and not self._faucetizer.validate_tail_acl_config(tail_acl):
+                error_msg = 'DVA auth was disabled due to missing ACL for tail_acl config'
```
Comment (grafnu) [inline forch/forchestrator.py:226]: We need to be consistent here.  Is "DVA" an actual feature of Forch or is it a collection of features?  If it's a collection, then this error message should be explicit about what is actually disabled.

File: forch/faucetizer.py [hunk 2]
```diff
@@ -225,13 +225,15 @@ def _initialize_host_ports(self, behavioral_faucet_config):
     def _finalize_host_ports_config(self, behavioral_faucet_config, new_testing_device_vlans):
         testing_port_vlans = new_testing_device_vlans.values()
         testing_port_configured = False
+        has_tail_acl = self._config.tail_acl and self._has_acl(self._config.tail_acl)
```
Comment (grafnu) [inline forch/faucetizer.py:228]: THis doesn't seem consistent with the validate function you created.  Why not just use the validate function here?  Should the check for self._config.tail_acl also be in the validate function?

**Review by grafnu — COMMENTED** [review]
bit confused because I thought we were going with the "if a tail_acl is not configured then it will use a default "empty" acl instead" to avoid the reconfiguration problem... ?  Either way, just checking that this WAI and does what is needed.

**Comment by didovesei** [comment]
> bit confused because I thought we were going with the "if a tail_acl is not configured then it will use a default "empty" acl instead" to avoid the reconfiguration problem... ? Either way, just checking that this WAI and does what is needed.

So after having discussed with Eng, we agreed that the existence of tail_acl configuration will be validated during config generation. Forch only needs to handle the case where the tail_acl is configured but no corresponding ACL is present.

**Comment by grafnu** [comment]
Got it, sounds good, thanks.

File: forch/forchestrator.py (hunk 1 above)
Comment (didovesei) [reply forch/forchestrator.py]: Update the message to "All auth was disabled...".

File: forch/faucetizer.py (hunk 2 above)
Comment (didovesei) [reply forch/faucetizer.py]: Done

File: forch/faucetizer.py [hunk 3]
```diff
@@ -122,9 +122,9 @@ def process_device_behavior(self, eth_src, behavior, static=False):
 
             self.flush_behavioral_config()
 
-    def validate_tail_acl_config(self, tail_acl):
+    def validate_tail_acl_config(self):
```
Comment (grafnu) [inline forch/faucetizer.py:125]: validate is a verb but this function doesn't do anything.  tail_acl_config_valid()?

File: forch/faucetizer.py [hunk 4]
```diff
@@ -225,13 +225,15 @@ def _initialize_host_ports(self, behavioral_faucet_config):
     def _finalize_host_ports_config(self, behavioral_faucet_config, new_testing_device_vlans):
         testing_port_vlans = new_testing_device_vlans.values()
         testing_port_configured = False
+        should_apply_tail_acl = self._config.tail_acl and self.validate_tail_acl_config()
```
Comment (anurag6) [inline forch/faucetizer.py:228]: Correct me if I have this wrong:
A = self._config.tail_acl; B = self._has_acl(self._config.tail_acl); A' = not self._config.tail_acl;

validate_tail_acl_config = A' or B

should_apply_tail_acl = A and (A' or B)
                                      = (A and A') or (A and B)
                                      = None or (A and B).  = A and B

Logically, would it be the same thing to simplify should_apply_tail_acl = self._config.tail_acl and self._has_acl(self._config.tail_acl)

Comment (anurag6) [inline forch/faucetizer.py:228]: NIT: I think 'should_' is redundant

Comment (anurag6) [reply forch/faucetizer.py:228]: My bad. Just checked comments above.

**Review by anurag6 — APPROVED** [review]

File: forch/faucetizer.py (hunk 3 above)
Comment (didovesei) [reply forch/faucetizer.py]: right.. Done

File: forch/faucetizer.py (hunk 4 above)
Comment (didovesei) [reply forch/faucetizer.py]: I see..

Comment (didovesei) [reply forch/faucetizer.py]: Done

**Review by grafnu — APPROVED** [review]

**Pull request merged**